Example/wifi_sta/firmware/src/app.c

```diff
@@ -75,7 +114,176 @@ uint8_t CACHE_ALIGN work[SYS_FS_FAT_MAX_SS];
 
 
 /* TODO:  Add any necessary local functions.
-*/
+ */
+
+
+bool validateUserPass (char * user, char * pass){
+    
+    if ((strlen(user) > 10) || (strlen(pass) > 10)) {
+        
+        SYS_CONSOLE_MESSAGE("Username or Password is too long\n\r");
+        return false;
+    }
+
+    return true;
+}
+
+bool FTP_Command_Initialize(void){
+    
+    if (!SYS_CMD_ADDGRP(ftpCmdTbl, sizeof(ftpCmdTbl)/sizeof(*ftpCmdTbl), "ftp", ": commands"))
+    {
+        SYS_ERROR(SYS_ERROR_ERROR, "Failed to create FTP Commands\r\n");
+        return false;
+    }
+}
+
+static inline void _Command_setECC(SYS_CMD_DEVICE_NODE* pCmdIO, int argc, char** argv){
+    
+    setECC = true;                          //ECC will be used to store FTP server login credentials
+    appData.state = APP_ECC_Init;
+}
+
+static int _Command_AddUser(SYS_CMD_DEVICE_NODE* pCmdIO, int argc, char** argv){
+    
+    if(setECC){
+        clientLoginInfo client;
```

Comment:
This is a local variable and the buffer's inside can have a uninitialized values, now if you do strncpy it doesnt guarantee that the destination buffer gets a string termination char '\0' at the end of the string. the validateuserPass can find a string of length more than given length.

Example, if the user passes a 5 bytes string (username or password) then if the local variable has a non zero (which is very common) values after 5 bytes, the string length would be different than the passed 5 bytes.

Reply:
removed the function